update readme and example bundle (#15)
deislabs/porter-docker: examples/docker-mixin-test/README.md

# Docker Mixin Example Bundle

This is an example bundle to show how to use the Docker mixin in Porter!

# Contents

## porter.yaml

This is the porter manifest. See https://porter.sh/author-bundles/ for 
details on every field and how to configure your bundle. This is a required
file. 

The commands available in the docker mixin are docker pull, push, build, 
run, remove, and login. 

> ### Notes on docker login

Reviewer: You've written great content here. How about we link to it from the readme in the docker login command so that people know to go here. e.g. "See an example for how to use docker login and securely provide your username and password"

Reviewer 2: +1.  Either a "check out the example bundle and its README for a closer look at mixin functionality" somewhere in the main README, or, like Carolyn suggests, linking specifically to this section from the `login` part of the main README.

Login has two optional parameters - username and password. To pass in the 

> username and password, you can either enter them in the yaml or set them 

Reviewer: Let's leave out "you can set them in the yaml" as that isn't what we recommend, embedding sensitive credentials in the yaml. So the text should instead just send them down the recommended path.

> Login has two optional parameters - username and password. See an [example] (../../README.md) for how to use 
> docker login and securely provide your username and password. To pass in the 

Reviewer: This link should be from the readme to this section.

```suggestion
Login has two optional parameters - username and password. To pass in the 
```

as environment variables and pass them in as credentials. In order to pass 
them in as credentials, you need to add the credentials section in the yaml. 

> You need to run porter credentials generate and specify where the values come

Reviewer: At the moment, it's kind of hard to find some of these commands and know what to do with them so maybe we should link to them.

```suggestion
You need to run [porter credentials generate](https://porter.sh/cli/porter_credentials_generate/) and specify where the values come
```

> from. Then, when you run porter install, pass in the credentials by doing 

Reviewer: ```suggestion
from. Then, when you run [porter install](https://porter.sh/cli/porter_install), pass in the credentials by doing 
```

> --cred or -c followed by the name of the credential set.

Reviewer: Let's show them an example of what the final command will look like
```suggestion
--cred or -c followed by the name of the credential set, for example `porter install --cred mycreds`.
```

## Dockerfile-cookies
This dockerfile was created to show how you can use docker build within a 

> bundle. Docker build under install uses this dockerfile to print out a message.

Reviewer: It wasn't quite clear to me at first what you were referencing, maybe something like this would help?

```suggestion
bundle. The docker build step defined in the install action of the porter.yaml uses this dockerfile to print out a message.
```
